staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -179,20 +179,41 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		return &roundTripper{a, rt}
 	}
 
-	getCert := c.TLS.GetCert
-	c.TLS.GetCert = func() (*tls.Certificate, error) {
-		// If previous GetCert is present and returns a valid non-nil
-		// certificate, use that. Otherwise use cert from exec plugin.
-		if getCert != nil {
-			cert, err := getCert()
-			if err != nil {
-				return nil, err
+	if t, ok := c.Transport.(*http.Transport); ok && t != nil {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		getCert := t.TLSClientConfig.GetClientCertificate
+		t.TLSClientConfig.GetClientCertificate = func(hi *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+			// If previous GetCert is present and returns a valid non-nil
+			// certificate, use that. Otherwise use cert from exec plugin.
+			if getCert != nil {
+				cert, err := getCert(hi)
+				if err != nil {
+					return nil, err
```

Comment:
would we want to check the exec plugin in an error case?

Reply:
Makes sense to me, added.
It's an unlikely configuration, but doesn't hurt.